Service/src/auth/auth.py

```diff
@@ -8,20 +8,30 @@ def login_required(f):
    @wraps(f)
    def decorator(*args, **kwargs):
       try: 
-         authHeader = request.headers.get('Authorization').split(' ')
-         if len(authHeader) != 2 or authHeader[0] != "token":
-            raise Exception('Invalid token')
-         authToken = authHeader[1]
-         g.cursor.execute("SELECT * FROM User WHERE token = %s", authToken)
-         g.user = g.cursor.fetchone()
-         if g.user is None:
-            raise Exception('Invalid token')
+         validate_session()
       except Exception as e:
          return make_response(jsonify(NotLoggedIn=errors['notLoggedIn']),401)
       
       return f(*args, **kwargs)
-      
    return decorator
 
+
+def validate_session():
+   authHeader = request.headers.get('Authorization').split(' ')
+   if len(authHeader) != 2 or authHeader[0] != "token":
+      raise Exception('Invalid token')
+   authToken = authHeader[1]
+   g.cursor.execute("SELECT * FROM User WHERE token = %s", authToken)
```

Comment:
The argument `authToken` needs to be either a list/tuple/dict or else character escaping isn't guaranteed. Simply change it to `(authToken,)` to force it to be a tuple. [Here's](https://github.com/PyMySQL/PyMySQL/blob/master/pymysql/cursors.py#L116) the commented source code about it ([here](https://github.com/PyMySQL/PyMySQL/blob/master/pymysql/cursors.py#L151) is the `execute` method if you want to follow the logic). Right now these queries are left open to SQL injection and should be fixed asap.

The [library docs](https://github.com/PyMySQL/PyMySQL#example) also show a single param arg with the tuple wrapping if you want to see an example.

Reply:
hm, asI remember PyMySQL should handle injection for you...I'll take a look later

Reply:
Looking at the source code it looks like it attempts to do it. Probably safe, maybe it's an artifact of an older version. Idk why they still do it in their examples.